HC-127 Add recap site

sampoder opened the pull request.

sampoder committed:
nuke a lot of stuff, and some stuff

sampoder committed:
Add Facts

sampoder committed:
Adjust Column Sizes

lachlanjc commented:
#### 📝Changed routes:


| [**/**](https://summer-git-sunset.hackclub.dev/) | [**/landing**](https://summer-git-sunset.hackclub.dev/landing) |
|:-:|:-:|
|<a href="https://summer-git-sunset.hackclub.dev/"><img src="https://images.weserv.nl?url=https://s.microlink.io/CBkfDfgKjdaaSnAGgBgut1qZHHc_.png&w=600" alt="Screenshot of /" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/aKJPNZEhN7GjYFdVBuss8jbdEz0_.png">(view full size)</a> |<a href="https://summer-git-sunset.hackclub.dev/landing"><img src="https://images.weserv.nl?url=https://s.microlink.io/mivMGGHk3rXVJ8_NnrUbSm7uTEA_.png&w=600" alt="Screenshot of /landing" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/tOS_lUxIZceWnwNEkjv8l96gqSQ_.png">(view full size)</a> |



| [**/prelaunch**](https://summer-git-sunset.hackclub.dev/prelaunch) |
|:-:|
|<a href="https://summer-git-sunset.hackclub.dev/prelaunch"><img src="https://images.weserv.nl?url=https://s.microlink.io/WeGzX5qwkWYFjU6sPDrsuaxmY_M_.png&w=600" alt="Screenshot of /prelaunch" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/H7TNkUbPysyVkIX3Dlj6xrWDrpg_.png">(view full size)</a> |







Commit 8c7c6125575504d442dbcc858e8ee7aa90533ba8 (https://summer-git-sunset.hackclub.dev).

sampoder commented:
wow good mooorrrning Lachlan bot

sampoder committed:
Start Hardware Grants Page

sampoder committed:
Add Stuff so I can show Rishi

sampoder committed:
Add Actual Stats

sampoder committed:
Mix up the packages

sampoder committed:
Add Hardware Intro (written by Chris)

lachlanjc reviewed (changes requested):
Hey! So I'd like to keep the existing site around, & make the recap page just an addition. Maybe bring back the components & do `git mv pages/index.js pages/landing.js` & then we'll have the recap site at the homepage of Summer?

sampoder commented:
okk I will unnuke

sampoder committed:
un nuke

sampoder committed:
Try to fix Mentions

sampoder committed:
Landing in the wrong folder :(

sampoder committed:
change coz react told me @lachlanjc is this right?

sampoder commented:
ok ik Smith is broken, investigating rn

sampoder committed:
Add first draft of Shoulder Bot page

sampoder commented:
<img width="825" alt="Screenshot 2020-07-31 at 10 18 53 PM" src="https://user-images.githubusercontent.com/39828164/89043963-e551c180-d37b-11ea-997d-2ef6ef4b2f4e.png">
<img width="880" alt="Screenshot 2020-07-31 at 10 19 49 PM" src="https://user-images.githubusercontent.com/39828164/89044008-f4d10a80-d37b-11ea-87fa-53fe3a6e1522.png">


ummmm whattt putting this issue on hold for now coz I'm thinking of ideas sorry

sampoder committed:
pause lanfing page stuff

sampoder committed:
add more things

lachlanjc commented:
Yeah, so if we upgrade to Next 9.5, `revalidate` is now stable, so you remove the `unstable_` prefix. More here: https://nextjs.org/blog/next-9-5

sampoder commented:
Awesome, thanks for sharing with me!

sampoder committed:
Fix up landing page and add some more stuff

sampoder committed:
Hover Effects

sampoder committed:
Add AMA Page and Arts Week Start

sampoder committed:
Add Connect Part

sampoder committed:
Shorten Connect Copy

sampoder committed:
Update AMA.js

sampoder committed:
Add a scrappy community page (to be improved)

sampoder committed:
Update scrapbook.js

sampoder committed:
Revert "Update scrapbook.js"

This reverts commit 96f12f000b27ddc492729aac806957906a2aeec8.

sampoder committed:
Make things for interactive and remove community events section

sampoder committed:
Remove clutter

sampoder committed:
uncommited stuff

sampoder committed:
Spacing fix

sampoder committed:
Change header copy

really only adapting it

sampoder committed:
Add inital scrapbook copy

Adapted from https://notebook.lachlanjc.com/2020-07-30_how_scrapbook_works/

Co-Authored-By: Lachlan Campbell <lachlan@hackclub.com>

sampoder committed:
Random stuff

sampoder committed:
Ensure proper credit for everyone who helped out with Summer of Making, update a url and add placeholder gif

sampoder committed:
im not lachlan

sampoder committed:
credit lachlan for other site tho

sampoder committed:
Add actual image (sorry minions)

sampoder commented:
short term things to do:

- [x] fix up the view more button, looks stupid and weird and just out of place
- [x] add an led themed border around the album cover project
- [x] mobile fixes lol rip me
- [x] iPad fixes rip me again
- [ ] make the arts week thing more interesting looks very stale
- [ ] find a 4th project
- [x] add Phantom Sans
- [ ] Fix heading on mobile
- [ ] sizing

long term things to do:

- [ ] get the video content set up AUGUST 17th 

things to think about:

- [ ] more moving things
- [ ] gradients?

sampoder committed:
Add LED Animation Thing

sampoder committed:
View More Button Changed

sampoder committed:
Spice up the Arts Part

sampoder committed:
Revert "Spice up the Arts Part"

This reverts commit 6d2cda1aac6325293426dee67b10a451811f3f9f.

sampoder committed:
Update scrapbook.js

sampoder committed:
Massive Mobile Improvements

sampoder commented:
for sam on mobile: https://summer-cvzczeyuq.hackclub.dev

sampoder commented:
current issues on mobile (not simulated): map text and Limor video

sampoder commented:
correction: issues with chrome support, oof

sampoder committed:
chrome fixes

sampoder commented:
https://summer-hqireb06w.hackclub.dev/

sampoder committed:
Add Content and update Meta

sampoder committed:
Update _app.js

sampoder committed:
Make the switch: Choose Phantom Sans

sampoder committed:
Fix margins

Introduced by font change

sampoder committed:
Add Highlights

Code from hackclub.com/clubs

Co-Authored-By: Lachlan Campbell <lachlan@hackclub.com>

sampoder committed:
Font Size fixes

sampoder committed:
Making things look better on mobile :)

sampoder committed:
Nuke dark mode fullly

sampoder committed:
Shuffle

sampoder commented:
dark mode:

![](https://s.yimg.com/ny/api/res/1.2/nQCAEHEI0r.FW9Vv4LybZg--/YXBwaWQ9aGlnaGxhbmRlcjt3PTcwNTtoPTUyOS4yODU3MTQyODU3MTQz/https://s.yimg.com/uu/api/res/1.2/VZburN609.ACTi0gf0EWNA--~B/aD03NDE7dz05ODc7c209MTthcHBpZD15dGFjaHlvbg--/https://media.zenfs.com/en/the_national_interest_705/1fae267970ec65c96104520803de6ab5)

sampoder committed:
Compressed gif

lachlanjc commented:
Shared my thoughts on Slack already. What's your timeline/idea of rolling this out? I think it makes sense to wait a week or two but the stats might be out of date by then. Then do you want it at summer. or what are you thinking there?

sampoder commented:
I'm also thinking to wait a week or two, I can update the stats when it comes to that time

sampoder committed:
Update map.js

sampoder committed:
Update shoulderbot.js

sampoder committed:
Update arts.js

sampoder committed:
Update arts.js

sampoder committed:
Change Shoulderbot to Polaroid || Also add nav bar

@lachlanjc I added nav bar so you can make it home screen

changed project coz the other isnt ready for demo :/

sampoder committed:
Update polaroid.js

lachlanjc commented:
@sampoder Can you update this branch reflecting the new homepage changes on the /landing on this branch?

sampoder commented:
Sure thing I'll do it tmw

sampoder commented:
Also fyi im making the videos this weekend was giving people as much time as possible to make their project

lachlanjc reviewed (changes requested):
As I said, this is so amazing, I’m so excited to have a homepage made by a community member!!! Just reviewed the site, some things to fix:

- [x] Can you move all the components for the page into a `components/recap` folder for organization?
- [x] Fix typo in intro text block: “It’s was a theme”
- [x] Let’s move the hardware partner logos to under the hardware stats, & more obviously credit them with some text as well (did it in a different way)
- [x] What if we embed the YouTube video instead of the photo of the camera, & drop the button? Here’s code for it: https://github.com/hackclub/v3/commit/951fdfdf2973da287c4900b75b405c73aba892ce [NOTE: to save on data I am using a gif instead]
- [x] Can you put in the filler images?
- [x] Let’s drop the second orange project section unless you have an awesome project to put there?
- [x] I want to add a very obvious credit to you near the top :)
- [x] Let’s link “Scrapbook” in the 3rd center text block
- [x] Add the Nicky AMA
- [x] Add a “See more on YouTube” link on the AMAs section that links to our YT channel [NOTE: not supper happy with the AMA page right now]
- [x] Let’s switch the “Join Slack” button to `variant=“cta”`
- [x] Let’s change “Throughout the summer, we hosted events for Hack Clubbers to connect with one another.” to more specific copy about what kind of events we actually ran
- [x] Host all images ourselves, not using imgur [NOTE: temporary project ones have not been switched]

After you pop through these, I’ll do a final design pass, sound good?

sampoder commented:
off the top of my head these all sound good, bit worried about the data YouTube of YouTube though will like construct a real sentence tomorrow

sampoder committed:
Move components into /recap

sampoder committed:
Fixed typo

sampoder committed:
b

sampoder commented:
first struggle, not sure what to do with the collab was thinking something like this with the text properly aligned vertically and replacing the second thank you.

issues are I don't think it looks great its cramped and it looks even worse on mobile

will look into alternatives later on (stored this on a separate local branch for now)


<img width="1341" alt="Screenshot 2020-08-22 at 7 52 15 AM" src="https://user-images.githubusercontent.com/39828164/90943430-66b4e500-e44c-11ea-9d85-6bd871a796de.png">

sampoder committed:
Revert "b"

This reverts commit 8d3a868904a28c70a09f9bcb55ec9e3a4c890936.

sampoder committed:
more stuff

lachlanjc commented:
Cool. Feel free to leave there & I can take a pass on Monday

sampoder commented:
lol @lachlanjc GitHub lets me edit your comment for some reason I'm editing it to make it a check list :)

lachlanjc commented:
Go for it!

sampoder committed:
Link the AMAs, add Nicky Case and link to YT channel

sampoder committed:
Update connect.js

sampoder committed:
Fix LEDMusic, redo Connect Copy

sampoder committed:
Move to Hack Club CDN

sampoder committed:
add a very obvious credit to me

i mean this might be too obvious

sampoder committed:
Sync up

sampoder committed:
Revert "Revert "b""

This reverts commit 00ab3ee51043c4a0d90c0226f9412d9a5db40414.

sampoder committed:
add fancy github thing

sampoder commented:
oh oof, I forgot smth

sampoder committed:
fix

sampoder committed:
other fix

sampoder committed:
more changes

sampoder committed:
nuke use sound

sampoder committed:
cool back grounds

sampoder committed:
:egg: easter egggg

sampoder committed:
kinda working rewind

sampoder committed:
attempt to fix vercel redirect

sampoder commented:
wow only one bug here + one styles issue I already knew about, fantastic!

sampoder commented:
jeeezzz this all increased

<img width="158" alt="Screenshot 2020-08-23 at 9 51 33 PM" src="https://user-images.githubusercontent.com/39828164/90979940-d29d6780-e58a-11ea-8537-40a34a2d77f7.png">

sampoder commented:
whhhaaattt the styling bug is only on my computer... YESSSS I LIKE ME RED BUTTONS

sampoder committed:
small fixes..

sampoder commented:
ya this happens some times idk why tho, just gotta redeploy

sampoder committed:
Redeploy me!

sampoder committed:
for codesandboz

sampoder committed:
Update theme.css

sampoder committed:
blind push!

going to go downstairs, when i come up this shoudl be looking amazing

sampoder committed:
to infinity and beyond!

sampoder committed:
quick fix

sampoder committed:
first draft switching to video

- not wide enough
- is the wrong video, need to work out mux

sampoder committed:
:egg: another one

sampoder committed:
organise my life

sampoder committed:
Revert "first draft switching to video"

This reverts commit 7be82017a4988f03806be7e2a79677d09de637aa.

lachlanjc committed:
Fix conflicts

lachlanjc committed:
Upgrade dependencies

lachlanjc committed:
Fix Intro warning

lachlanjc committed:
Iterate on design

sampoder committed:
become a better digital citizen

sampoder committed:
Merge branch 'sunset' of https://github.com/hackclub/summer-site into sunset

sampoder committed:
fixing up the merge stuff

sampoder committed:
Add rel="noopener best practise

sampoder committed:
fixing up the SEO!

sampoder committed:
Let's get that SEO score rising!

sampoder committed:
Add alt text

sampoder committed:
Improve accessibility

sampoder committed:
redeploy!

sampoder committed:
does this sorta thing improve scores?

sampoder committed:
redeploy idk why this keeps happening

sampoder committed:
Compressed Images from last night

sampoder committed:
redeploy coz lighthouse stats not showing

sampoder committed:
just going to yolo it!

sampoder commented:
what is this r error?

sampoder committed:
let's give it a go!

sampoder committed:
Update AMA.js

sampoder committed:
Trying more stuff :)

sampoder committed:
Revert "Trying more stuff :)"

This reverts commit 9c68b36a6a82f385b152bc04e21664816cd53a01.

sampoder committed:
Revert "Update AMA.js"

This reverts commit d63a2045989fdb68116b6e711a23dea0b64a0452.

sampoder committed:
Revert "let's give it a go!"

This reverts commit bb772ad0b40868bd5b6771bd9d757124ef3581c7.

sampoder committed:
Revert "Revert "let's give it a go!""

This reverts commit 7fd3f8cf9653d099b34da83ec4d14b60be6963c7.

sampoder committed:
maybe?

sampoder committed:
fix image, im stuck

sampoder committed:
starting with mux!

sampoder committed:
Move Recap Site to use Videos (#58)

* Update stat.js

* Revert "first draft switching to video"

This reverts commit 7be82017a4988f03806be7e2a79677d09de637aa.

* Revert "Revert "first draft switching to video""

This reverts commit 91e60fdad3da03e50c99e5d35ca80ed5282d552d.

* Revert "Revert "Revert "first draft switching to video"""

This reverts commit 9c1ef5e2c42e9398ee1cebf6e67c29a4d99e6f96.

* Hopefully these show up

* fix deploy issues

sampoder committed:
fingers crossed

sampoder commented:
this could go horribly wrong

sampoder committed:
import

sampoder committed:
Update connect.js

sampoder committed:
lighthouse broken

lachlanjc commented on components/recap/mentionFancy.js:
```diff
@@ -30,7 +30,7 @@ const Mention = memo(({ username }) => {
   return (
     <a sx={{textDecoration:'none'}} href={`https://scrapbook.hackclub.com/${username}`}>
-      <a sx={{textDecoration:'none'}} className="mention post-text-mentionfancy">
+      <a className="mention post-text-mentionfancy" href={`https://scrapbook.hackclub.com/${username}`}>
```
`<a>` tags should never be nested—I think you can remove the higher-level one entirely. & fyi, `sx` will be ignored here since you’re not using Theme UI’s jsx pragma—you’ll need to use their `Link` component (you can do `import { Link as StyledLink }` or something) to use Theme UI styling

sampoder replied:
ngl, I have no clue what this piece of code does.....

sampoder replied:
I was just too scared to delete it

lachlanjc commented on components/recap/thankyou.js:
```diff
@@ -20,13 +20,16 @@ import {
 } from 'theme-ui'
 
+import usePrefersMotion from './use-prefers-motion'
+import useHasMounted from './use-has-mounted'
```
These hooks are already available in `lib/`, want to avoid duplicating

lachlanjc commented on components/recap/thankyou.js:
```diff
@@ -20,13 +20,16 @@ import {
 )
 
-export default () => (
+const SheetOLd = () => (
```
Is this component being used?

sampoder replied:
nope, I'm leaving it there for now in case I need to reference something from it.. I'll put it into a file on my computer

lachlanjc commented on components/recap/use-has-mounted.js:
```diff
@@ -0,0 +1,12 @@
+// Full credit to https://joshwcomeau.com/snippets/react-hooks/use-has-mounted
```
See above comment, already in `lib/`

lachlanjc commented on components/recap/video.js:
```diff
@@ -0,0 +1,53 @@
+    <video
+      ref={videoRef}
+      poster={`https://image.mux.com/${mux}/thumbnail.jpg?width=512&fit_mode=pad&time=0`}
+      className="post-attachment"
```
```suggestion
```

lachlanjc commented on pages/index.js:
```diff
@@ -1,11 +1,15 @@
+  { loading: () => <p>...</p> }
+)
 import MIDI from '../components/recap/midi'
 import LedMusic from '../components/recap/ledMusic'
```
This isn’t a big deal, but the fact that you mix `formattingLike`, `thisand`, `like-this` across your components is :)

sampoder replied:
... unique

sampoder replied:
on the other side they are all quite nice tho, im just saying

lachlanjc commented on components/recap/thankyou.js:
```diff
@@ -91,3 +94,152 @@ export default () => (
+        id="slack"
+        sx={{ overflow: 'hidden', position: 'relative' }}
+      >
+        <Box
```
You can remove this if you don’t have a gradient overlay

lachlanjc commented on components/recap/thankyou.js:
```diff
@@ -91,3 +94,152 @@ export default () => (
+          muted
+          loop
+          playsInline
+          poster="slack-poster.png"
```
Set this to the URL of the image of the first frame

lachlanjc commented on components/recap/thankyou.js:
```diff
@@ -91,3 +94,152 @@ export default () => (
+            width: '100%',
+            zIndex: -1
+          }}
+        >
```
You prob want to use Mux component here instead but with `autoPlay muted loop playsInline` & this styling

lachlanjc reviewed (commented):
You’re doing amazing work!!!

sampoder committed:
Switch to actual videos

sampoder committed:
redeploy

sampoder committed:
hmm

sampoder committed:
wow. this is actually looking nice

sampoder committed:
Update components/recap/video.js

Co-authored-by: Lachlan Campbell <lachlan@hackclub.com>

sampoder committed:
risky move

sampoder committed:
Merge branch 'sunset' of https://github.com/hackclub/summer-site into sunset

sampoder committed:
oldie

sampoder committed:
lazy load images

sampoder committed:
Revert "oldie"

This reverts commit 9cbc4c555a0b25b874201fda9a2cc0f24a930982.

sampoder committed:
im so confused... i think it's safari's fault but like whyyy

sampoder committed:
blah balh

sampoder committed:
good bye!

sampoder committed:
Revert "Revert "oldie""

This reverts commit 35ec1e3916073e5255a468e00a78deb001543a29.

sampoder committed:
switch to mux and fill it up!

sampoder committed:
devtools != editor

sampoder committed:
Update thankyou.js

sampoder committed:
Little changes

sampoder committed:
More small improvements

sampoder committed:
blind push! well kinda! but all new AMA part!

sampoder committed:
small little improvement!

sampoder committed:
mmmm

sampoder committed:
Make them interactive

sampoder committed:
Remove whit space!

sampoder committed:
Last few things

sampoder committed:
Fix up paths

sampoder commented:
wow it might actually be ready...

sampoder commented:
FYI: I'm getting these scores on Lighthouse


<img width="1440" alt="Screenshot 2020-08-30 at 10 05 57 PM" src="https://user-images.githubusercontent.com/39828164/91661194-fde50100-eb0c-11ea-8529-ac64e72d2bc4.png">



<img width="1440" alt="Screenshot 2020-08-30 at 10 05 11 PM" src="https://user-images.githubusercontent.com/39828164/91661174-e1e15f80-eb0c-11ea-903b-e88f679b798b.png">

sampoder committed:
Reduce Network Payload

sampoder commented:
Network amount

<img width="750" alt="Screenshot 2020-08-30 at 10 10 05 PM" src="https://user-images.githubusercontent.com/39828164/91661275-911e3680-eb0d-11ea-934e-a9049ed58470.png">

sampoder committed:
Update scrapbook.js

sampoder committed:
Update hardwaregrants.js

sampoder committed:
Update scrapbook.js

lachlanjc committed:
Rename /pre-launch

lachlanjc committed:
Iterate site

lachlanjc committed:
Continue iterating

lachlanjc committed:
Continue iterating design

lachlanjc committed:
Try to fix deploy issue
